benchmark848/benchmark.cpp

```diff
@@ -0,0 +1,348 @@
+}
+
+/* Delete Node Group Test */
+void DeleteNodeGroupTest(const unique_ptr<Connection> &connection) {
+    string tableName = RandomSelectTable("");
+    string nextTableName = RandomSelectTable(tableName);
+
+    /*
+     * 1. Create table here first
+     *      Checkpoint will create physical storage
+     */
+    CreateTable(connection, tableName);
+    Checkpoint(connection);
+
+    /*
+     * 2. Delete table entries based on random ratio
+     *      chunks are recycled in Checkpoint.
+     */
+    int beginId = (rand() % 50) * NUM_ROWS / 100;
+    int endId = (rand() % 50 + 50) * NUM_ROWS / 100;
+    DeleteEntries(connection, tableName, beginId, endId);
+    Checkpoint(connection);
+
+    /*
+     * 3. Create a new table to resue recycled spaces
+     */
+    CreateTable(connection, nextTableName);
+    Checkpoint(connection);
+
+    /* 4. Drop both tables here to wrap up the test */
+    DropTable(connection, tableName);
+    DropTable(connection, nextTableName);
+    Checkpoint(connection);
+}
+
+int main(int argc, char* argv[])
+{
+    if (argc == 1) {
+        cout<<"Please type following input:"<<endl
+            <<"    -N <number of iteration you want to run>"<<endl
+            <<"    -D <directory that saves the csv source file>"<<endl;
+        return 0;
+    }
+
+    /* parse the argument first */
+    char *csvFileDir = GetCmdOption(argv, argv + argc, "-D");
+    if (csvFileDir == nullptr) {
+        cout<<"Please use -D to specify the directory that saves the csv source file"<<endl;;
+        return 0;
+    }
+    UpdateTableCopyQuery(csvFileDir);
+
+    int maxIteration = 0;
+    char *iterationsArg = GetCmdOption(argv, argv + argc, "-N");
+    if (iterationsArg == 0) {
+        cout<<"Please use -N to specify the directory that saves the csv source file"<<endl;
+        return 0;
+    }
+    maxIteration = std::atoi(iterationsArg);
+    cout<<"User Parameters:"<<endl
+        <<"    Csv file source directory: "<<csvFileDir<<endl
+        <<"    Max test iterations: "<<maxIteration<<endl;
+
+    /* Create an empty on-disk database and connect to it */
+    kuzu::main::SystemConfig systemConfig;
+    auto database = make_unique<Database>("testDB", systemConfig);
+    auto connection = make_unique<Connection>(database.get());
+
+    if (connection != nullptr) {
+        int curIter = 0;
+        while (curIter < maxIteration) {
+            cout << "---------------------------\nBegin " + to_string(curIter) + "th iterations\n---------------------------" << endl;
+            TestType testCase = (TestType)(rand() % 3);
+            switch (testCase) {
+                case DROP_TABLE: {
+                    /* ERICTODO: Calculate average checkpoint time here */
+                    cout << "Test Type: DROP_TABLE" << endl;
+                    DropTableTest(connection);
+                    break;
+                }
+                case ALTER_TABLE: {
+                    /* ERICTODO: Calculate average checkpoint time here */
```

Comment:
In case this helps:

The checkpoint function could be modified to return the running time, which we can then use to compute average times. Consider the following:
```
#include <chrono>

std::chrono::nanoseconds checkpoint(const unique_ptr<Connection> &connection) {
    // Get the starting timepoint
    auto start = std::chrono::high_resolution_clock::now();

    // run the checkpoint query
    connection->query("CHECKPOINT;");

    // Get the ending timepoint
    auto end = std::chrono::high_resolution_clock::now();

    // Calculate the duration in nanoseconds
    auto duration_ns = std::chrono::duration_cast<std::chrono::nanoseconds>(end - start);

   return duration_ns;
}
```
The duration_ns may be printed or arithmetic operations may be performed, using which we should be able to estimate average runtime.